src/zenml/integrations/seldon/steps/seldon_deployer.py

```diff
@@ -36,10 +43,49 @@
 )
 from zenml.steps.step_context import StepContext
 from zenml.utils import io_utils
+from zenml.utils.materializer_utils import save_model_metadata
+from zenml.utils.source_utils import import_class_by_path
 
 logger = get_logger(__name__)
 
 
+class CustomDeployParameters(BaseModel):
+    """Custom model deployer step extra parameters.
+
+    Attributes:
+        predict_function: Path to Python file containing predict function.
+
+    Raises:
+        ValueError: If predict_function is not specified.
+
+    Returns:
+        predict_function: Path to Python file containing predict function.
+    """
+
+    predict_function: str
+
+    @validator("predict_function")
+    def predict_function_validate(cls, predict_func_path: str) -> str:
+        """Validate predict function.
+
+        Args:
+            predict_func_path: predict function path
+
+        Returns:
+            predict function path
+
+        Raises:
+            ValueError: if predict function path is not valid
+        """
+        if not predict_func_path:
+            raise ValueError("Predict function path is required.")
+        try:
+            import_class_by_path(predict_func_path)
+        except AttributeError:
```

Comment:
Similar to kserve, this could also check that the function is a function/callable to catch errors early